### ISSUE TITLE
bug: IonPopover size "auto" does not match content 

### ISSUE THREAD

**Issue by Facj** [ionic-team/ionic-framework#27877]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

[According to the documentation](https://ionicframework.com/docs/api/popover#css-custom-properties), IonPopover width will match popover content if type "auto" is selected in size attribute.
However, it forces width to 250px, the default size described in [popover-md-width](https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/popover/popover.md.vars.scss) and does not adapt to wider content.

### Expected Behavior

IonPopover width should adapt to content width when attribute size = "auto".

### Steps to Reproduce

I have created a minimal app with three different pages that can be accessed using buttons in Home page. All of them include IonPopover or IonSelect with popover interface. Popover width does not match wide content in any of them.

Page1 includes a popover that can be opened using a button.
Page 2 include a form with IonSelect fields with popover interface.
Page 3 is the same as Page 2 but the form in nested inside a more complex page content, that matchs closely my real app.

### Code Reproduction URL

https://github.com/Facj/ionpopover-issue

### Ionic Info

$ ionic info

Ionic:

   Ionic CLI       : 6.6.0 (C:\Users\fcastro.NORVENTO\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework : @ionic/react 7.2.1

Utility:

   cordova-res : not installed
   native-run  : 1.7.2

System:

   NodeJS : v16.15.1 (C:\Program Files\nodejs\node.exe)
   npm    : 9.4.2
   OS     : Windows 10

### Additional Information

There are quite a few related questions in Ionic forum ([1](https://forum.ionicframework.com/t/not-able-to-override-the-width-of-ion-select-dropdown-ionic-6/230763), [2](https://forum.ionicframework.com/t/ion-select-not-wide-enough/152346), ) and Stackoverflow ([1](https://stackoverflow.com/questions/70538484/ionic-select-option-text-cutting-off), [2](https://stackoverflow.com/questions/41754966/ionic-2-ion-select-width), [3](https://stackoverflow.com/questions/38622159/how-to-set-ion-select-component-with-100-width-in-ionic-2/67971384#67971384) ) trying to set IonPopover size. Users are looking for CSS-based solutions, which would not be necessary if IonPopover size "auto" would actually match content width.

**Comment by liamdebeasi** [comment]
Thanks for the report. Can you clarify how you are trying to set the size of the popover? By default `ion-popover` uses a width of 250px. However, you can set `--width: auto` on the `ion-popover` host to have the popover be sized relative to the size of the child content.

I tried setting the following CSS in your app, and the popover was sized based on the size of its content:

```
ion-popover {
  --width: auto;
}
```

**Comment by Facj** [comment]
Thanks! It does work as you say.

I was doing nothing to enforce popover size based on content. Reading [the documentation](https://ionicframework.com/docs/api/popover#properties), I assumed auto width was the default and, therefore, I thought this was a bug.

The actual behavior does make sense but the documentation is not clear. 

![imagen](https://github.com/ionic-team/ionic-framework/assets/5441615/7ab8ef28-f160-468a-80ae-68ad67e5ee0e)

**Comment by liamdebeasi** [comment]
I agree that this is confusing. I opened a PR to clarify this: https://github.com/ionic-team/ionic-framework/pull/27894

I am going to close this since this is not a bug in Ionic, but let me know if you have any questions.

**Comment by Facj** [comment]
Thanks for your help!

Going over the popovers in my code, I have realized `--width: auto` is not correctly applied when the list of options is long and scroll is needed. I have updated [my minimal example](https://github.com/Facj/ionpopover-issue) to include a long list of options in Page2.

**Comment by liamdebeasi** [comment]
Seems to be working on my end: 
![image](https://github.com/ionic-team/ionic-framework/assets/2721089/23de5a6f-dc98-4b6a-9d60-6fc1d91fd043)


Setting `--width: auto` causes the popover to have a width of ~265px. Without the `--width: auto` it reverts back to its platform default of 250px.

**Comment by Facj** [comment]
On my end, there is a scroll in the popover and it seems to avoid the `--width: auto `enforcement.

![imagen](https://github.com/ionic-team/ionic-framework/assets/5441615/2beef7f4-eb47-45b5-8e3e-cfc4a50401be)

Is it possible that your screen is big enough to fit all the options so no scroll is needed? Could you test a smaller screen size?

**Comment by Facj** [comment]
Yes, it looks like the scrolllbar is the cause of the truncation. 

It seems like  `width: auto` is not applied anymore when the scrollbar appears because width is back to 250px.
